### PR TITLE
Test it

### PR DESCRIPTION
**Pull request by MOJOliciousFTW**
## Description

This pull request introduces a full solution pipeline for the Kaggle Playground Series S5E8 bank term deposit prediction challenge. It adds a comprehensive script for data loading, exploration, preprocessing, model training (using several machine learning algorithms), ensemble prediction, and submission file creation. Additionally, it updates the project dependencies in `pyproject.toml` to support the new code.

**Machine learning pipeline and data handling:**

* Added a complete script in `main.py` that loads the training and test datasets, performs exploratory data analysis (EDA) with visualizations, preprocesses features (including encoding and feature engineering), trains multiple models (Logistic Regression, Random Forest, XGBoost, LightGBM), compares their AUC scores, and generates ensemble predictions and submission files.

**Project dependencies:**

* Updated `pyproject.toml` to include required dependencies for the solution: `lightgbm`, `matplotlib`, `numpy`, `pandas`, `scikit-learn`, `seaborn`, and `xgboost`. This ensures all libraries used in the script are available for the project.

## Additional information
- [ ] Issues this PR closes are linked under development --->
- This PR is related to issue: 

## Types of changes
- [ ] We use labels --->

## Checklist:
- [ ] Works on my PC!

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -1,12 +1,347 @@
...
+    # Simple average ensemble
+    ensemble_pred = np.mean(list(test_predictions.values()), axis=0)
+
+    # Weighted ensemble (giving more weight to better performing models)
+    weights = {
+        "logistic_regression": 0.2,
+        "random_forest": 0.25,
+        "xgboost": 0.3,
+        "lightgbm": 0.25,
+    }
```
[the remark this answers is not in the document]
Comment (MOJOliciousFTW) [reply main.py:279]: they are magic.
